config/manager-base/manager.yaml

```diff
@@ -26,15 +26,20 @@ spec:
     spec:
       affinity:
         nodeAffinity:
-          requiredDuringSchedulingIgnoredDuringExecution:
-            nodeSelectorTerms:
-            - matchExpressions:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 1
+            preference:
+              matchExpressions:
               - key: node-role.kubernetes.io/control-plane
                 operator: Exists
-            - matchExpressions:
+          - weight: 1
+            preference:
+              matchExpressions:
               - key: node-role.kubernetes.io/master
                 operator: Exists
-            - matchExpressions:
+          - weight: 1
+            preference:
+              matchExpressions:
               - key: kmm.node.kubernetes.io/control-plane
                 operator: Exists
```

Comment:
With your change, this section become redundant. Please remove this `nodeAffinity`'s entry completely as now we don't need to manually label nodes with `kmm.node.kubernetes.io/control-plane` to make it work on "special environments".